src/apps/Altinn.AccessManagement/src/Altinn.AccessManagement.Api.Enduser/Controllers/ConnectionController.cs

```diff
@@ -22,281 +23,209 @@ namespace Altinn.AccessManagement.Api.Enduser.Controllers;
+public class ConnectionController(IHttpContextAccessor accessor, IEnduserConnectionService connectionService, IDbAudit audit) : ControllerBase
 {
     private IHttpContextAccessor Accessor { get; } = accessor;
 
+    private IEnduserConnectionService ConnectionService { get; } = connectionService;
+
+    private IDbAudit Audit { get; } = audit;
+
     /// <summary>
-    /// Creates an assignment between the authenticated user's selected party and the specified target party.
+    /// Get connections between the authenticated user's selected party and the specified target party.
     /// </summary>
-    /// <param name="party">The GUID identifying the party the authenticated user is acting on behalf of.</param>
-    /// <param name="from">The GUID identifying the party the authenticated user is acting for</param>
-    /// <param name="to">The GUID identifying the target party to which the assignment should be created.</param>
-    /// <param name="cancellationToken"><see cref="CancellationToken"/></param>
     [HttpGet]
+    [DbAudit(Claim = AltinnCoreClaimTypes.PartyUuid, System = AuditDefaults.EnduserApiStr)]
     [Authorize(Policy = AuthzConstants.POLICY_ACCESS_MANAGEMENT_ENDUSER_READ)]
-    [ServiceFilter(typeof(AuthorizePartyUuidClaimFilter))]
-    [ProducesResponseType<AssignmentExternal>(StatusCodes.Status200OK, MediaTypeNames.Application.Json)]
-    [ProducesResponseType<ProblemDetails>(StatusCodes.Status400BadRequest, MediaTypeNames.Application.Json)]
+    [ProducesResponseType<PaginatedResult<AssignmentExternal>>(StatusCodes.Status200OK, MediaTypeNames.Application.Json)]
+    [ProducesResponseType<AltinnProblemDetails>(StatusCodes.Status400BadRequest, MediaTypeNames.Application.Json)]
     [ProducesResponseType(StatusCodes.Status401Unauthorized)]
     [ProducesResponseType(StatusCodes.Status403Forbidden)]
-    public async Task<IActionResult> GetConnection([FromQuery] Guid party, [FromQuery] Guid? from, [FromQuery] Guid? to, CancellationToken cancellationToken = default)
+    public async Task<IActionResult> GetConnections([FromQuery] ConnectionInput connection, [FromQuery, FromHeader] PagingInput paging, CancellationToken cancellationToken = default)
     {
-        if (!from.HasValue && !to.HasValue)
+        var userUuid = Audit.Value;
+        if (ValidationRules.EnduserGetConnection(userUuid.ChangedBy, connection.Party, connection.From, connection.To) is var problem && problem is { })
         {
-            return BadRequest();
+            return problem.ToActionResult();
         }
 
-        if (!(from.HasValue && from.Value == party) && !(to.HasValue && to.Value == party))
+        var result = await ConnectionService.Get(from: connection.From.ConvertToUuid(userUuid.ChangedBy), to: connection.To.ConvertToUuid(userUuid.ChangedBy), cancellationToken: cancellationToken);
```

Comment:
Autentisert bruker skal ikke brukes direkte som from eller to i spørringer.

Det skal alltid gis spesifikk input for Party, From og To som er det som skal brukes i oppslagene.